fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -889,6 +1010,38 @@ where
     }
 }
 
+impl<S, Tx, Ecal> Interpreter<S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    /// Executes `Upload` transaction without initialization VM and without invalidation
+    /// of the last state of execution of the `Script` transaction.
+    ///
+    /// Returns `Upload` transaction with all modifications after execution.
+    pub fn upload(
+        &mut self,
+        tx: Ready<Upload>,
+    ) -> Result<Upload, InterpreterError<S::DataError>> {
+        self.verify_ready_tx(&tx)?;
+
+        let (_, checked) = tx.decompose();
+        let (mut upload, metadata): (Upload, <Upload as IntoChecked>::Metadata) =
+            checked.into();
+        let base_asset_id = *self.base_asset_id();
+        let gas_price = self.gas_price();
+        Self::upload_inner(
```

Comment:
What's the purpose of having an `inner` function here. And it's so far from the "outer", despite having the same `impl` bounds AFAICT. There are a few places in the code I've noticed that do this and it deepens my context stack while reading the code and makes it harder to read IMO.

I'm a big fan of keeping functions short, but I also like keeping the context as local as possible and making things declarative as possible so you can just read the function signature to get enough context as possible. This code just makes me wonder, what's the difference between `upload` and `upload_inner`.

I suggested adding a `upload_bytecode_subsection` method above. Maybe with the combination of that as well as a `get_bytecode`, etc, you could fit this all in the "outer" `upload` function while keeping things short.

Reply:
It is related to [this](https://github.com/FuelLabs/fuel-vm/issues/251).

<img width="1216" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/8e78c527-8f9a-4ec5-b207-ef9c7de4fcb1">


The idea is that the `run` function should only work with the `Script` type since you can run only script transactions. All other types of transactions don't require `Interpreter`'s features or context.

I'm keeping each section separately because I hope that in the future, we will refactor `Interpreter`=D